### ISSUE TITLE
onDestroyView and RetainViewMode.RELEASE_DETACH

### ISSUE THREAD

**Issue by mradzinski** [bluelinelabs/Conductor#353]
Hi, quick question over here:

Based on the Readme lifecycle diagram, whenever the parent activity hits `onStop()` (for example, when the app is sent to background by pressing the home button) and the RetainViewMode is set to it's default `RetainViewMode.RELEASE_DETACH` I should expect to hit `onDetach()` and `onDestroyView()`, right?

I'm having this situation where if I set the retain mode to be `RetainViewMode.RELEASE_DETACH`, `onDestroyView()` is only being invoked when a controller is being popped/pushed to backstack or when I destroy the activity, but when the parent activity hits `onStop()` I only hit `onDetach()` and never `onDestroyView()`. So, I don't know if I'm misinterpreting the diagram or maybe the part where it says "on Activity stopped" is wrong.

**Comment by bejibx** [comment]
I think the "on Activity stopped"-part only refers to onDetach event.
> onDestroyView() is only being invoked when a controller is being popped/pushed to backstack or when I destroy the activity, but when the parent activity hits onStop() I only hit onDetach() and never onDestroyView().

I think this is intended behaviour. There is no point to destroy views on every Activity#onStop() call.

**Comment by mradzinski** [comment]
I thought actually `RETAIN_DETACH` was designed for that purpose. If I set it then cool, I shouldn't hit `onDestroyView` after a detach, but if I'm setting the controller to release its view once it gets detached then I should expect that whenever it hits `onDetach` then the view should also be destroyed. If that's the intended behavior then it should be stated in the README as an exception to the rule.

**Comment by bejibx** [comment]
Hm, readme says
> setRetainViewMode can be called on a Controller with one of two values: RELEASE_DETACH, which will release the Controller's view as soon as it is detached from the screen (saves memory), or RETAIN_DETACH, which will ensure that a Controller holds on to its view, even if it's not currently shown on the screen (good for views that are expensive to re-create).

Now I'm confused too.

**Comment by mradzinski** [comment]
Indeed. Actually, I just tested `RETAIN_DETACH` on a controller, pushed another one and the one with `RETAIN_DETACH` hit the `onDestroyView` and it's view was recreated upon reattaching. That (as per the documentation) shouldn't happen at all.

**Comment by Gaket** [comment]
For me , `RETAIN_DETACH ` works as intended when I push a controller over an existing one using the same router. I checked it in source code and it looks like it shouldn't destroy view when activity is Stopped only, without being destroyed. Do you have the latest version of the lib?
